doc/source/whatsnew/v0.17.0.txt

```diff
@@ -56,8 +57,32 @@ Deprecations
 Removal of prior version deprecations/changes
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+.. _dask: https://dask.readthedocs.org/en/latest/
+
+.. _whatsnew_0170.gil:
+
+Releasing the GIL
+~~~~~~~~~~~~~~~~~
+
+We are releasing the global-interpreter-lock (GIL) on some cython operations.
+This will allow other threads to run simultaneously during computation, potentially allowing performance improvements
+from multi-threading. Notably ``groupby`` and some indexing operations are a benefit to this. (:issue:`8882`)
+
+For example the groupby expression in the following code will have the GIL released during the groups computation, e.g. ``df.groupby('key')``
```

Comment:
maybe say "during the factorization step" instead of "during the groups computation", not totally clear what the "groups" computation is